src/Features/CSharp/Portable/UseIsNullCheck/CSharpUseIsNullCheckForReferenceEqualsCodeFixProvider.cs

```diff
@@ -28,17 +28,22 @@ private static SyntaxNode CreateIsNullCheck(SyntaxNode argument)
                 (ExpressionSyntax)argument,
                 SyntaxFactory.ConstantPattern(SyntaxFactory.LiteralExpression(SyntaxKind.NullLiteralExpression))).Parenthesize();
 
-        private static SyntaxNode CreateIsNotNullCheck(SyntaxNode notExpression, SyntaxNode argument)
-            => ((PrefixUnaryExpressionSyntax)notExpression).WithOperand((ExpressionSyntax)CreateIsNullCheck(argument));
+        private static SyntaxNode CreateIsNotNullCheck(SyntaxNode argument)
+        {
+            return SyntaxFactory
+                .BinaryExpression(
+                    SyntaxKind.IsExpression,
+                    (ExpressionSyntax)argument,
+                    SyntaxFactory.PredefinedType(SyntaxFactory.Token(SyntaxKind.ObjectKeyword)))
+                .Parenthesize();
+        }
 
         protected override SyntaxNode CreateNullCheck(SyntaxNode argument, bool isUnconstrainedGeneric)
             => isUnconstrainedGeneric
-                ? CreateEqualsNullCheck(argument, SyntaxKind.EqualsExpression)
+                ? CreateEqualsNullCheck(argument)
                 : CreateIsNullCheck(argument);
 
         protected override SyntaxNode CreateNotNullCheck(SyntaxNode notExpression, SyntaxNode argument, bool isUnconstrainedGeneric)
```

Comment:
isUnconstrainedGeneric no longer used.  consider removing from the entire inheritance hierarchy.

Reply:
Now simplified